include/network_element.hpp

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include "packet.hpp"
+#include "event.hpp"
+
+#include <unordered_map>
+#include <queue>
+#include <memory>
+
+class RoutingPacket;
+
+/**
+ * @brief Abstract network element (link, switch, host, etc.)
+ * 
+ */
+class NetworkElement {
```

Comment:
It seems to me that it would be better to define independent classes for a network device and a link separately, i.e. without inheritance from a common abstract class.